frontend/src/utils/analysis-hooks.ts

```diff
@@ -150,9 +148,19 @@ export const useAnalysisForm = (
     [adminLevel],
   );
 
-  const adminLevelLayerData = useSelector(
-    adminLevelLayer ? layerDataSelector(adminLevelLayer.id) : () => undefined,
-  ) as LayerData<BoundaryLayerProps> | undefined;
+  const boundaryDataResult = useBoundaryData(adminLevelLayer?.id || '');
+
+  const adminLevelLayerData: LayerData<BoundaryLayerProps> | undefined =
+    useMemo(() => {
+      if (!boundaryDataResult.data || !adminLevelLayer) {
+        return undefined;
+      }
+      return {
+        layer: adminLevelLayer,
+        data: boundaryDataResult.data,
+        date: Date.now(),
+      };
+    }, [boundaryDataResult.data, adminLevelLayer]);
```

Comment:
The function passed to `useMemo` should be pure, so it shouldn't really access the current time. But more generally, I don't think `useMemo` is needed at all here, because the calculation you're running isn't actually expensive.
I would try to go with the following instead, what do you think?
```suggestion
  const adminLevelLayerData: LayerData<BoundaryLayerProps> | undefined =
    boundaryDataResult.data && adminLevelLayer
      ? {
          layer: adminLevelLayer,
          data: boundaryDataResult.data,
          date: Date.now(),
        }
      : undefined;
```
(I think there are lots of places in the codebase where we could do the same thing, but that's a separate discussion)